### DISCUSSION TITLE
HSS hangs after USB device connection

### DISCUSSION THREAD

**Discussion by JV73** [polarfire-soc/polarfire-soc-documentation#181] [HSS and U-Boot] [answered]
Having trouble using the HSS to access eMMC over USB from a MAC running a Windows virtual machine.  This set-up worked previously with a different Libero/MSS/HSS build.  and the current Libero/MSS/HSS build is working on another kit with a Windows Laptop, so we're struggling to find a reason for this behavior: 

The HSS starts up normally and we can pre-empt it from booting by hitting a key.  We use 'MMC' and then 'USBDMSC' to enter the USB device mode; and the HOST VM/Windowns on the MAC does see the polarfire memory device come alive and it appears that the first 'block read' operation begins - but then the HSS appears to be 'hung'.  See below: 

![COM4 HHS hang](https://user-images.githubusercontent.com/106761965/185253563-7ce89ada-db3c-41f8-81ae-fab743ccdcaf.jpeg)

**Comment by hughbreslin** [comment]
Hi @JV73 

Is the spinning bar to the left of "0 bytes" keep moving and if you enter `CTRL-C` does it disconnect or do you mean it has fully crashed?

**Reply by JV73** [reply]
It's fully crashed.  The spinning stops and the console is non-responsive.  We have to instances that are experiencing this. 
One is a linux machine connected to an ICICLE-KIT-ES, and the other is a MAC running a Windows VM connected to a different ICICLE-KIT-ES.   My Windows 10 PC with the same Libero/HSS programming file is working.

**Comment by JV73** [comment, marked as answer]
We decided to re-build the Libero project & HSS with settings in the MSS configurator adjusted to match the 'expected' DDR partitioning in the Yocto build flow (32 MB DDRlo and 1888 MB DDRhi).  The updated project is working. It both the Windows PC set-up and the Linux PC set-up; we have yet to try it on the MAC with Windows VM.

So - the PolarFire Yocto scripts and/or the HSS expect the AMP memory partitioning do be as above.
